servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultMultiAddressUrlHttpClientBuilder.java

```diff
@@ -244,7 +267,184 @@ public StreamingHttpClient apply(final UrlKey urlKey) {
                 singleAddressInitializer.initialize(urlKey.scheme, urlKey.hostAndPort, builder);
             }
 
-            return builder.buildStreaming();
+            StreamingHttpClient singleClient = builder.buildStreaming();
+
+            return new SingleAddressStreamingHttpClientWrapper(singleClient, executionContext.executionStrategy());
```

Comment:
Instead of wrapping manually, consider using a filter that you can append before invoking `singleAddressInitializer`. That way you won't need to delegate/implement most of the methods, it should be enough to implement only `request` method.

Reply:
I don't understand how that would work.

Reply:
Implement a regular `StreamingHttpClientFilterFactory` that returns a `StreamingHttpClientFilter` as we usually do and append it before invoking an initializer.